alcs-frontend/src/app/features/application/post-decision/post-decision.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { combineLatestWith, tap } from 'rxjs';
+import { ApplicationDetailService } from '../../../services/application/application-detail.service';
+import { ApplicationReconsiderationDetailedDto } from '../../../services/application/application-reconsideration/application-reconsideration.dto';
+import { ApplicationReconsiderationService } from '../../../services/application/application-reconsideration/application-reconsideration.service';
+import { ToastService } from '../../../services/toast/toast.service';
+import { ConfirmationDialogService } from '../../../shared/confirmation-dialog/confirmation-dialog.service';
+import { BaseCodeDto } from '../../../shared/dto/base.dto';
+import { formatDateForApi } from '../../../shared/utils/api-date-formatter';
+import { PostDecisionDialogComponent } from './post-decision-dialog/post-decision-dialog.component';
+
+@Component({
+  selector: 'app-post-decision',
+  templateUrl: './post-decision.component.html',
+  styleUrls: ['./post-decision.component.scss'],
+})
+export class PostDecisionComponent implements OnInit {
+  fileNumber: string = '';
+  postDecisions: ApplicationReconsiderationDetailedDto[] = [];
+  codes: BaseCodeDto[] = [];
+
+  constructor(
+    public dialog: MatDialog,
+    private applicationDetailService: ApplicationDetailService,
+    private applicationReconsiderationService: ApplicationReconsiderationService,
+    private toastService: ToastService,
+    private confirmationDialogService: ConfirmationDialogService
+  ) {}
+
+  ngOnInit(): void {
+    this.applicationDetailService.$application
```

Comment:
Can you add a $destroy and a takeUntil so the subscribes is properly cleaned up?